xinterpreters/xinterpreters/base/base_interpreter.py

```diff
@@ -0,0 +1,308 @@
+        @param data  The blob of data to set the tensor to.
+        @param model_index The engine to target, for interpreters that support multiple models
+        running concurrently. Defaults to 0 for use with a single model.
+        """
+        return
+
+    @abstractmethod
+    def get_output_tensor(self, output_index=0, tensor=None, model_index=0) -> "Output tensor data":
+        """! Abstract for reading the data in the output tensor of a model.
+        @param output_index  The index of output tensor to target.
+        @param tensor Tensor of correct shape to write into (optional)
+        @param model_index The engine to target, for interpreters that support multiple models
+        running concurrently. Defaults to 0 for use with a single model.
+        @return The data that was stored in the output tensor.
+        """
+        return
+
+    @abstractmethod
+    def get_input_tensor(self, input_index=0, model_index=0) -> "Input tensor data":
+        """! Abstract for reading the data in the input tensor of a model.
+        @param output_index  The index of output tensor to target.
+        @param tensor Tensor of correct shape to write into (optional)
+        @param model_index The engine to target, for interpreters that support multiple models
+        running concurrently. Defaults to 0 for use with a single model.
+        @return The data that was stored in the output tensor.
+        """
+        return
+
+    @abstractmethod
+    def initialise_interpreter(self, model_index=0) -> None:
+        """! Abstract initialising interpreter with model associated with model_index.
+        @param model_index The engine to target, for interpreters that support multiple models
+        running concurrently. Defaults to 0 for use with a single model.
+        """
+        return
+
+    @abstractmethod
+    def invoke(self) -> None:
+        """! Abstract for invoking the model and starting inference of the current
+        state of the tensors
+        """
+        return
+
+    @abstractmethod
+    def close(self, model_index=0) -> None:
+        """! Abstract deleting the interpreter
+        @params model_index Defines which interpreter to target in systems with multiple
+        """
+        return  
+
+    @abstractmethod
+    def tensor_arena_size(self) -> "Size of tensor arena":
+        """! Abstract to read the size of the tensor arena required
+        @return size of the tensor arena as an integer
+        """
+        return
+
+    @abstractmethod
+    def _check_status(self, status):
+        """! Abstract to read a status code and raise an exception
+        @param status Status code
+        """
+        return
+
+    @abstractmethod
+    def print_memory_plan(self) -> None:
+        """! Abstract to print a plan of memory allocation
+        """
+        return
+
+
+    def get_input_tensor_size(self, input_index=0, model_index=0) -> "Size of input tensor":
```

Comment:
It's very good that we got rid of the tflite files. I'm not sure about the tflite package though, it seems to be unmaintained, https://github.com/jackwish/tflite. I think it's better to depend on tensorflow, which means we can clean up more code for getting input/output details etc.

Reply:
Have decided to not use tensorflow at the moment as it doesn't give us enough to add it as a dependency.